src/box/session_settings.c

```diff
@@ -342,12 +342,16 @@ session_settings_space_execute_update(struct space *space, struct txn *txn,
 
 	const char *new_key, *key = request->key;
 	uint32_t new_size, new_key_len, key_len = mp_decode_array(&key);
-	if (key_len == 0) {
-		diag_set(ClientError, ER_EXACT_MATCH, 1, 0);
+	if (key_len != 1) {
+		diag_set(ClientError, ER_EXACT_MATCH, 1, key_len);
+		index_set_error_details(pk_def);
+		error_set_key_with_part_count(key, key_len);
 		return -1;
 	}
-	if (key_len > 1 || mp_typeof(*key) != MP_STR) {
+	if (mp_typeof(*key) != MP_STR) {
 		diag_set(ClientError, ER_KEY_PART_TYPE, 0, "string");
+		index_set_error_details(pk_def);
+		error_set_key_with_part_count(key, key_len);
```

Comment:
Let's use `index_point_lookup_validate` here (in the previous patch).

Reply:
Done.

<details>
<summary>diff</summary>

```diff
--- a/src/box/session_settings.c
+++ b/src/box/session_settings.c
@@ -342,16 +342,8 @@ session_settings_space_execute_update(struct space *space, struct txn *txn,

        const char *new_key, *key = request->key;
        uint32_t new_size, new_key_len, key_len = mp_decode_array(&key);
-       if (key_len != 1) {
-               diag_set(ClientError, ER_EXACT_MATCH, 1, key_len);
+       if (index_point_lookup_validate(pk_def, key, key_len) != 0) {
                index_set_error_details(pk_def);
-               error_set_key_with_part_count(key, key_len);
-               return -1;
-       }
-       if (mp_typeof(*key) != MP_STR) {
-               diag_set(ClientError, ER_KEY_PART_TYPE, 0, "string");
-               index_set_error_details(pk_def);
-               error_set_key_with_part_count(key, key_len);
                return -1;
        }
        key = mp_decode_str(&key, &key_len);
```

</details>